lib/moab/storage_object_validator.rb

```diff
@@ -137,15 +128,11 @@ def check_data_directory(version_path, version, allow_content_subdirs = true)
     end
 
     def check_data_sub_dirs(version, data_sub_dirs)
+      return found_unexpected(data_sub_dirs, version, EXPECTED_DATA_SUB_DIRS) if data_sub_dirs.size > EXPECTED_DATA_SUB_DIRS.size
       errors = []
-      if data_sub_dirs.size > EXPECTED_DATA_SUB_DIRS.size
-        errors.concat found_unexpected(data_sub_dirs, version, EXPECTED_DATA_SUB_DIRS)
-      else
-        errors.concat missing_dir(data_sub_dirs, version, [METADATA_DIR]) unless data_sub_dirs.include?(METADATA_DIR)
-        errors.concat found_unexpected(data_sub_dirs, version, EXPECTED_DATA_SUB_DIRS) unless subset?(data_sub_dirs,
-                                                                                                      EXPECTED_DATA_SUB_DIRS)
-      end
-      errors
+      errors.concat missing_dir(data_sub_dirs, version, [METADATA_DIR]) unless data_sub_dirs.include?(METADATA_DIR)
+      return errors if data_sub_dirs.to_set.subset?(EXPECTED_DATA_SUB_DIRS.to_set)
+      errors.concat found_unexpected(data_sub_dirs, version, EXPECTED_DATA_SUB_DIRS)
```

Comment:
i think this reads a little easier with 

```ruby
errors.concat xxx unless/if
errors.concat xxx unless/if
errors
```

than

```ruby
errors.concat xxx unless/if
return errors if 
errors.concat xxx
```

in my opinion, the logic is clearer for the first.

Reply:
Yeah, I can see that.  Not great but whatever.  My point here was mainly to get rid of the single-line single-use `subset?` method in the class.